highfidelity/hifi
# Edit model dialog improvements

Dialog now allows selection of collision type and whether or not it should be dynamic.  Also fixed a lot of bad practices and code style errors in edit.js.

### Testing
1. Start up interface with edit.js running
2. Click the button to create a new model
3. When the "exact" collision mode is selected, the "dynamic" checkbox should be unclickable and a warning should show
4. The checkbox should remember its state before and after selecting "exact" mode
5. A model should not be created if the url is blank or if cancel is clicked
6. The URL doesn't have to be valid to create a model, it just can't be empty
7. Ensure selected properties are reflected in the actual entity created
8. Everything should work in HMD
9. Import a variety of mesh models and apply all the different hull configurations and then test that you see a result that you'd expect. The Whole Model and Sub-Mesh options don't create per triangle hulls so don't expect to get exact results. However, you should be able to see for models that are not spherical or square that a more complex hull is being created. There is not yet a way to visualize the hull in debug mode.
